lib/pbench/server/templates.py

```diff
@@ -348,14 +360,58 @@ def add_mapping(self, name: str, body: Dict[AnyStr, Any]):
         """
         self.mappings.add_mapping(name, body)
 
-    def resolve(self):
+    def update(self, template: Template):
         """
-        In the initial implementation, this just loads the template files.
+        Update the template object from the database model object.
 
-        Placeholder: this will check the template DB and decide whether we
-        need to load the JSON files from disk based on modification dates.
+        Args:
+            template: DB Template instance
+        """
+        self.modified = template.mtime
+        self.version = template.version
+        self.mappings = JsonFile.raw_json(template.mappings)
+        self.settings = JsonFile.raw_json(template.settings)
+        self.template_name = template.template_name
+        self.index_pattern = template.index_template
+
+    def resolve(self):
         """
+        Check the on-disk template mapping files' modification dates against
+        the templates DB. Only if the template's base index name doesn't exist
+        in the DB, or if the DB object is older than the on-disk template do
+        we load the mapping files from disk.
+        """
+        try:
+            template = Template.find(self.name)
+        except TemplateNotFound:
+            template = None
```

Comment:
Of the three calls, this saves me from a `None` check in the other two; this is the only case where failing to find the template in the DB is acceptable. (Because we'll then create it.)